File: backend/analytics-django/analytics/mock_data.py
MOCK_CANDIDATES = [
    {
        "type": "CONFIG_CHANGE",
        "key": "JWT_EXPIRY",
        "delta": "15m -> 5ms",
        "time_diff_seconds": 120,
        "confidence": 0.85
    },
    {
        "type": "DEPLOYMENT",
        "version": "v1.4.2",
        "actor": "ci-bot",
        "time_diff_seconds": 300,
        "confidence": 0.60
    },
    {
        "type": "FEATURE_FLAG",
        "flag": "new-auth-flow",
        "state": "ON",
        "time_diff_seconds": 1800,
        "confidence": 0.35
    }
]

File: backend/replay-engine-node/mockData.js
const getMockSnapshot = (timestamp) => ({
    time: new Date(new Date(timestamp).getTime() - 3600000).toISOString(), // 1 hour ago
    state: {
        config: { JWT_EXPIRY: '15m' }, // Initial state
        flags: { 'new-auth-flow': false }
    }
});

const getMockEvents = (timestamp) => [
    {
        type: 'CONFIG_CHANGE',
        timestamp: new Date(new Date(timestamp).getTime() - 1800000).toISOString(),
        schemaVersion: 1,
        payload: { key: 'JWT_EXPIRY', value: '5m' }
    },
    {
        type: 'FEATURE_FLAG',
        timestamp: new Date(new Date(timestamp).getTime() - 900000).toISOString(),
        schemaVersion: 1,
        payload: { flag: 'new-auth-flow', enabled: true }
    }
];

module.exports = {
    getMockSnapshot,
    getMockEvents
};
